### PR TITLE
Fix evaluation if no edisp is set

### LINKED ISSUE #5274: FluxPointsEstimator fails if no edisp is set
**Gammapy version**
present dev

**Bug description**

Running `FluxPointsEstimator` on a dataset without an `edisp` gives a very confusing error message `ValueError: Can only stack equivalent maps or cutout of the same map.`

**To Reproduce**

Set `edisp=None` for any dataset (with e_true=e_reco) and proceed with flux points estimation in multiple energy bands (normal use case).

**Expected behaviour**
The analysis fails (as expected) during the fitting stage if `energy_true` has a different binning from the `energy` axis with a more understandable error 
```
File ~/Gammapy-dev/gammapy/gammapy/datasets/evaluator.py:245, in MapEvaluator._edisp_diagonal(self)
    242 @lazyproperty
    243 def _edisp_diagonal(self):
    244     return EDispKernel.from_diagonal_response(
--> 245         energy_axis_true=self.edisp.axes["energy_true"],
    246         energy_axis=self.edisp.axes["energy"],
    247     )

AttributeError: 'NoneType' object has no attribute 'axes'
```
However, if the `energy_axis` and the `energy_axis_true` have same binning, the fit proceeds fine and then the very confusing error is raised during flux points estimation.

1. Either the flux estimator (and also the fit) should fail with a decent error message consistently for no edisp OR
2. the diagonal edisp can be adapted with the `energy_axis` here 
https://github.com/gammapy/gammapy/blob/e9b22cdff573768ac2ab1e2a4a18250071ba3b60/gammapy/datasets/evaluator.py#L243
to use the dataset energy axis.

### PR DESCRIPTION
**Pull request by QRemy**
Resolves #5274.
Better solution than #5308 :)

### CODE REVIEW

**Review by AtreyeeS — APPROVED** [review]
Thanks @QRemy ! This is a very clean solution.
Maybe it would still be useful to raise a warning in `MapDataset` if `edisp is None` ?

File: gammapy/estimators/points/tests/test_sed.py [hunk 1]
```diff
@@ -430,6 +430,29 @@ def test_run_map_pwl_reoptimize(fpe_map_pwl_reoptimize):
...
+    assert_allclose(actual, [1.081434, 0.91077, 0.922176], rtol=1e-3)
+
+    datasets[0].edisp = None
+
+    fp = fpe.run(datasets)
+    table = fp.to_table()
+    actual = table["norm"].data
+    assert_allclose(actual, [1.081434, 0.91077, 0.922176], rtol=1e-3)
+
+    datasets[0].models["source"].apply_irf["edisp"] = True
```
Comment (fabiopintore) [inline gammapy/estimators/points/tests/test_sed.py:451]: Shouldn't you add:
```    fp = fpe.run(datasets)
         table = fp.to_table()```
here?

**Review by fabiopintore — COMMENTED** [review]
Thanks @QRemy . It looks definitely better than the other solution. It looks good to me. I have only a small comment for the test.

File: gammapy/estimators/points/tests/test_sed.py (hunk 1 above)
Comment (QRemy) [inline gammapy/estimators/points/tests/test_sed.py:451]:
```suggestion
    datasets[0].models["source"].apply_irf["edisp"] = True
    fp = fpe.run(datasets)
    table = fp.to_table()
```

Comment (QRemy) [reply gammapy/estimators/points/tests/test_sed.py:451]: yes added

**Pull request merged**